apps/api/src/auth/auth.service.ts

```diff
@@ -426,4 +428,25 @@ export class AuthService {
           )
     }
   }
+
+  async deleteUser(keycloakId: string) {
+    const user = await this.personService.findOneByKeycloakId(keycloakId)
+
+    //Check and throw if user is a beneficiary, organizer or corporate profile
+    if (user?.beneficiaries?.length || user?.organizer || user?.companyId) {
+      throw new InternalServerErrorException(
+        'Cannot delete a beneficiary, organizer or corporate profile',
+      )
+    }
+
+    return this.authenticateAdmin()
```

Comment:
Personally, I am fine with this .then format.
But I think so far the common pattern is to use await.

Reply:
I made this decision because if, for some reason, the promise from Keycloak does not resolve successfully, I want to interrupt the chain and refrain from removing the user in the API database. I am unsure if this is possible when I use await for every promise.